### DISCUSSION TITLE
[FR] More strict business processes needed

### DISCUSSION THREAD

**Discussion by alxrMironov** [inventree/InvenTree#6534] [Ideas]
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

In short, the problem is - the staff is not always so responsible or\and well-intentioned, how is want to be.
Some steps can easily be sabotaged, ignored by mistake or evil will, it really doesnt matter.
The software must be able to handle these situations and\or provide minimum tools for detection and restoration.

Some steps can easy be sabotaged or ignored.
Examples:
1) Worker has assembled part in fact, but been too lazy or just forgot to allocate all stock, due substitute issues or conflicts as example. He can easily set "Allow incomplete" and you have no real way to track it.
Incomplete build order or stock from it, have no really any signs of its incompleteness.
Proposal: Restrict dangerous actions with special permission per account.

2) There is no consistent way to track the reason for stock status changes. Like, you can make it from a stock list with reason or from "edit item" with no ability to set reason.
Proposal: Add ability to restrict status change without comment with special setting or permission per account.

3) There is no strict way to handle NOTES. Notes is a very nice and flexible way to handle any additional information, except... it can easily be erased in one click and you have no chance to check if there was some.
Proposal: Save full notes changes history or atleast track change events .

4) Sales\Build orders\Purchaes completeon.
Some actions cant be easily reverted after the completion. The BO as example never reverts his allocation status.
But there is many obivious reasons why some not prepared personel may interact with it and make mistakes sometime.
Proposal: Divide rights needed for BO\SO\PO(ant etc) edition and completion. Allow accountsto edit but not complete orders. 

### Do you want to develop this?

- [ ] I want to develop this.

**Comment by matmair** [comment]
Not sure a feature request (FR) is the right format for this, I can see at least 7 or 8 different FR's in the text right now, some of them are already filed.

**Comment by SchrodingersGat** [comment]
@alxrMironov each of these request is a significant task to implement and should be best tracked as separate issues. As @matmair calls out, please:

1. Check if each issue is already filed and you can add comments to that issue if so
2. If not already existing, create a new *specific* issue for each one. 

It is too hard to track multiple large issues in a single discussion!

**Comment by alxrMironov** [comment]
> @alxrMironov each of these request is a significant task to implement and should be best tracked as separate issues. As @matmair calls out, please:
> 
> 1. Check if each issue is already filed and you can add comments to that issue if so
> 2. If not already existing, create a new _specific_ issue for each one.
> 
> It is too hard to track multiple large issues in a single discussion!

It is not FR as it is (you just have no "discussion" type of issue).
Im want to know developers opinion and explain my.

**Reply by matmair** [reply]
I have moved this over to discussions. We use issues as task lists so they are not really the right thing to discuss multiple ideas. Here we have the option to discuss specific things in threads

**Comment by wolflu05** [comment]
> It is not FR as it is (you just have no "discussion" type of issue).

There are discussions, but they are separate from issues. You can find them on the repository page where code, issues, pull requests, ... are under the name discussions
